kbncontent.go

```diff
@@ -0,0 +1,314 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+// Package kbncontent implements routines for analyzing Kibana content.
+//
+// It provides information about Kibana assets and a single source of truth for what things are legacy and should no longer be used.
+package kbncontent
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/stretchr/objx"
+)
+
+// VisualizationDescriptor describes a visualization.
+// In this case, "visualization" means anything which can be embedded in a dashboard.
+type VisualizationDescriptor struct {
+	Doc             map[string]interface{}
+	SavedObjectType string
+	Link            string
+}
+
+func (v VisualizationDescriptor) findDocumentPathsAsString(paths []string) string {
+	m := objx.Map(v.Doc)
+
+	for _, path := range paths {
+		if m.Get(path).IsStr() {
+			return m.Get(path).Str()
+		}
+	}
+
+	return ""
+}
+
+// Type returns the root-level visualization type
+// currently empty for Lens
+func (v VisualizationDescriptor) Type() string {
+	if v.SavedObjectType != "visualization" {
+		return ""
+	}
```

Comment:
Not sure if this is a rare condition but either way might be better to return an error here and catch it at the higher level for _them_ to use as desired.

Reply:
In this case I think the "quiet" empty string return makes sense because that is how the original scripts functioned and matches the expectations of the legacy vis tracker. 

I've documented the behavior in the godoc above the function.

> // currently empty for Lens

In the future, it probably makes sense to remove this gating and support this method for Lens, too. I think we should re-evaluate this suggestion at that point.